src/crypto/secretbox/xsalsa20poly1305.rs

```diff
@@ -71,6 +71,25 @@ pub fn seal(m: &[u8],
     c
 }
 
+/// `seal_detached_inplace()` encrypts and authenticates a message `m` using a secret key `k` and a
+/// nonce `n`.  `m` is encrypted in place, so after this function returns it will contain the
+/// ciphertext.  The detached authentication tag is written to `mac`.
+pub fn seal_detached_inplace(m: &mut[u8],
```

Comment:
I'm not really happy with the API here. I think an api of the form
```
pub fn seal_detached_inplace(m: &mut [u8], &Nonce(ref n): &Nonce, &Key(ref k): &Key) -> Tag;
```

Where `Tag` is a newtype. This would be much more in spirit with the rest of the apis in sodiumoxide.

That would of course need to be complemented by a
```
pub fn seal_detached(m: &[u8], &Nonce(ref n), &Key(ref k)) -> (Vec<u8>, Tag);
```

I would like to deprecate the old `seal/open` apis in the future and only expose the new ones (renamed to seal/open).